dashboard/settings.py

```diff
@@ -438,11 +442,20 @@ def show_debug_toolbar(request):
 if "CSP" in ENV:
     MIDDLEWARE_CLASSES += ['csp.middleware.CSPMiddleware',]
     for csp_key, csp_val in ENV.get("CSP").items():
-        globals()["CSP_"+csp_key] = csp_val
+        # If there's a value set for this CSP config, set it as a global
+        if (csp_val):
+            globals()["CSP_"+csp_key] = csp_val
 # If CSP not set, add in XFrameOptionsMiddleware
 else:
     MIDDLEWARE_CLASSES += ['django.middleware.clickjacking.XFrameOptionsMiddleware',]
 
+# These are mostly needed by Canvas but it should also be in on general 
+CSRF_COOKIE_SECURE = ENV.get("CSRF_COOKIE_SECURE", True)
+if CSRF_COOKIE_SECURE:
```

Comment:
Does those variables get default values set if CSRF_COOKIE_SECURE is false?

Reply:
They just aren't set to anything. They [have defaults as part of Django](https://docs.djangoproject.com/en/1.11/ref/settings/).
[CSRF_TRUSTED_ORIGINS](https://docs.djangoproject.com/en/1.11/ref/settings/#std:setting-CSRF_TRUSTED_ORIGINS) = [] 
[SESSION_COOKIE_SECURE](https://docs.djangoproject.com/en/1.11/ref/settings/#session-cookie-secure)=False
[SECURE_PROXY_SSL_HEADER](https://docs.djangoproject.com/en/1.11/ref/settings/#secure-proxy-ssl-header) = None